api/src/infrastructure/docker.rs

```diff
@@ -491,6 +538,74 @@ impl DockerInfrastructure {
         Ok(())
     }
 
+    async fn fetch_existing_volumes(app_name: &String) -> Result<Vec<VolumeInfo>, ShipLiftError> {
+        let docker = Docker::new();
+        docker.volumes().list().await.map(|volume_infos| {
+            volume_infos
+                .into_iter()
+                .filter(|s| {
+                    s.labels
+                        .as_ref()
+                        .map(|label| label.get(APP_NAME_LABEL) == Some(app_name))
+                        .unwrap_or(false)
+                })
+                .collect::<Vec<VolumeInfo>>()
+        })
+    }
+
+    async fn create_docker_volume(
+        app_name: &str,
+        config: &ServiceConfig,
+    ) -> Result<String, ShipLiftError> {
+        let docker = Docker::new();
+        let volumes = docker.volumes();
+
+        let mut labels: HashMap<&str, &str> = HashMap::new();
+        labels.insert(APP_NAME_LABEL, app_name);
+        labels.insert(SERVICE_NAME_LABEL, config.service_name());
+
+        let volume_options = VolumeCreateOptions::builder().labels(&labels).build();
+        volumes
+            .create(&volume_options)
+            .await
+            .map(|volume_info| volume_info.name)
+    }
+
+    async fn create_host_config_binds(
+        app_name: &str,
+        image_info: Option<&ImageInfo>,
+        existing_volume: &[VolumeInfo],
+        config: &ServiceConfig,
+    ) -> Result<Vec<String>, ShipLiftError> {
+        let mut host_binds = Vec::new();
+
+        let attached_volume = match image_info {
+            Some(image) => image.attached_volume(),
```

Comment:
If we think of the aforementioned configuration approach, we should move the logic which volumes need to created for each service into the `DeploymentUnitBuilder`. This builder should enrich `DeployableService` with a list of volumes to be mounted. 

Later we can extend that in the hooks sections to be modifiable when users of PREvant want to share persistent data across services.